nncf/tensorflow/quantization/algorithm.py

```diff
@@ -194,7 +183,17 @@ def get_transformation_layout(self, model: tf.keras.Model) -> TFTransformationLa
     def build_controller(self, model: tf.keras.Model) -> 'QuantizationController':
         return QuantizationController(model, self.config, self._op_names)
 
+    def _parse_init_params(self):
+        self._batchnorm_adaptation = 'batchnorm_adaptation' in self.config.get('initializer', {})
+        self._range_init_params = self._parse_range_init_params()
+
+    def _parse_range_init_params(self) -> TFRangeInitParams:
+        range_init_params = extract_range_init_params(self.config)
+        return TFRangeInitParams(**range_init_params) if range_init_params is not None else None
+
     def initialize(self, model: tf.keras.Model) -> None:
+        self._parse_init_params()
```

Comment:
You moved parsing parameters into initialize function. If initialization config is not correct than the user will see error relative to incorrect config only on initialization step. BTW this is the last step of the quantization pipeline. I would recomend to validate config into __init__ method.

Reply:
I agree, thanks. Fixed.